src/PIL/ImageGrab.py

```diff
@@ -61,10 +61,13 @@ def grab(bbox=None, include_layered_windows=False, all_screens=False, xdisplay=N
                 left, top, right, bottom = bbox
                 im = im.crop((left - x0, top - y0, right - x0, bottom - y0))
             return im
-        elif shutil.which("gnome-screenshot"):
+        elif shutil.which("gnome-screenshot") or shutil.which("scrot"):
             fh, filepath = tempfile.mkstemp(".png")
             os.close(fh)
-            subprocess.call(["gnome-screenshot", "-f", filepath])
+            if shutil.which("scrot"):  # use scrot when have
+                subprocess.call([shutil.which("scrot"), "-z", "--overwrite", filepath])
```

Comment:
```suggestion
                subprocess.call([shutil.which("scrot"), "-z", filepath])
```
I don't think it should be necessary to overwrite the file, since it is a temporary file.

Reply:
Regarding this issue, in my local operating environment, if the --overwrite option is removed, the following exception message will be thrown:

PIL.UnidentifiedImageError: cannot identify image file
The reason for this is that the temporary file does not exist. However, after adding the --overwrite option, this issue disappears.

After testing, scrot defaults to renaming the file if it detects that the file to be saved already exists, and the --overwrite option can disable this feature and force the screenshot information to be written to the target file.